pkg/apiserver/experiment/experiment.go

```diff
@@ -67,11 +71,50 @@ func (s *Service) listExperiments(c *gin.Context) {}
 // TODO: need to be implemented
 func (s *Service) getExperimentDetail(c *gin.Context) {}
 
-// TODO: need to be implemented
-func (s *Service) deleteExperiment(c *gin.Context) {}
+func (s *Service) deleteExperiment(c *gin.Context) {
+	kind := c.Query("kind")
+	if kind == "" {
+		util.JSONEmptyData(c, statuscode.ParamNotSet, "kind param should be set")
+		return
+	}
+
+	ctx := context.TODO()
+	chaosKey := types.NamespacedName{Namespace: c.Param("ns"), Name: c.Param("name")}
+
+	kindsMap := map[string]runtime.Object{
+		v1alpha1.KindPodChaos:     &v1alpha1.PodChaos{},
+		v1alpha1.KindNetworkChaos: &v1alpha1.NetworkChaos{},
+		v1alpha1.KindStressChaos:  &v1alpha1.StressChaos{},
+		v1alpha1.KindIOChaos:      &v1alpha1.IoChaos{},
+		v1alpha1.KindKernelChaos:  &v1alpha1.KernelChaos{},
+		v1alpha1.KindTimeChaos:    &v1alpha1.TimeChaos{},
+	}
+	var (
+		chaos runtime.Object
+		ok    bool
+	)
+	if chaos, ok = kindsMap[kind]; !ok {
+		util.JSONEmptyData(c, statuscode.UndefinedKind,
+			fmt.Sprintf("kind %s is not supported", kind))
+		return
+	}
+	if err := s.kubeCli.Get(ctx, chaosKey, chaos); err != nil {
+		util.JSONEmptyData(c, statuscode.GetResourcesWrong,
+			"failed to get chaos; "+err.Error())
+		return
+	}
+
+	if err := s.kubeCli.Delete(ctx, chaos, &client.DeleteOptions{}); err != nil {
```

Comment:
Should we also need to delete the experiment in the DB? Or do you want to archive this experiment in db after deleting the experiment?

Reply:
I think deleting the current experiment in the cluster and deleting experiments in the databases should be 2 APIs because only archived experiments are in the database.

But I am not very sure. Maybe @cwen0 has some other ideas. WDYT?

Reply:
Oh, I mean, after deleting the experiment from the cluster, we need to archive the experiment to the db.

Reply:
Sorry I misunderstand your question before. But in another PR I already implemented the archive functionality. So when one experiment is deleted it will be added to dB immediately